plugins/lookup/aws_secret.py

```diff
@@ -130,7 +163,14 @@ def run(self, terms, variables, **kwargs):
                 if 'SecretString' in response:
                     secrets.append(response['SecretString'])
             except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-                raise AnsibleError("Failed to retrieve secret: %s" % to_native(e))
+                if e.response['Error']['Code'] == 'ResourceNotFoundException' and missing in ['warn', 'skip']:
+                    if missing == 'warn':
+                        self._display.warning('Skipping, did not find secret %s' % term)
+                elif e.response['Error']['Code'] == 'AccessDeniedException' and denied in ['warn', 'skip']:
+                    if denied == 'warn':
+                        self._display.warning('Skipping, access denied for secret %s' % term)
+                else:
+                    raise AnsibleError("Failed to retrieve secret: %s" % to_native(e))
```

Comment:
We also have a helper is_boto3_error_code (in module_utils.core).  This would give you an alternative  of:
```
except is_boto3_error_code('ResourceNotFoundException'):
    if missing == 'error':
        raise AnsibleError("Failed to find secret %s" % term)
    elif missing == 'warn':
        self._display.warning('Skipping, did not find secret %s' % term)
except is_boto3_error_code('AccessDeniedException'):
    if missing == 'error':
        raise AnsibleError("Failed to access secret %s (Permission Denied)" % term)
    elif missing == 'warn':
        self._display.warning('Skipping, access denied for secret %s' % term)
except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
    raise AnsibleError("Failed to retrieve secret: %s" % to_native(e))
```

Personally I find this logic a little easier to follow but does duplicate the raise

Reply:
i changed it to the helper